src/worker/predictive-text-studio-worker-impl.ts

```diff
@@ -17,20 +18,32 @@ const defaultCopyright = "";
 function doNothing() {
   // intentionally empty
 }
+import { keyboardDataObj } from "./models";
 
 export class PredictiveTextStudioWorkerImpl
   implements PredictiveTextStudioWorker {
-  private storage: Storage;
-
-  constructor(storage = new Storage()) {
-    this.storage = storage;
+  constructor(
+    private storage = new Storage(),
+    private keymanApi = new KeymanApi()
+  ) {
+    this.getLanguageData();
   }
 
   async updateBCP47Tag(bcp47Tag: string): Promise<void> {
     await this.storage.updateBCP47Tag(bcp47Tag);
     return this.generateKMPFromStorage();
   }
 
+  async getLanguageData(): Promise<void> {
+    await this.keymanApi
+      .fetchLanaguageData()
+      .then((languages: keyboardDataObj[]) => {
```

Comment:
Why do you use both `await` and Promise `.then` callbacks that `await` replaces?

Reply:
nice catch, only one await is needed here